implementation/src/main/java/io/smallrye/config/ConfigMappingProvider.java

```diff
@@ -1065,6 +1069,32 @@ private static Set<String> additionalMappedProperties(final Set<String> mappedPr
             mappedProperties.remove(propertyName);
         }
 
+        Set<String> envRoots = new HashSet<>();
+        for (String root : roots) {
+            envRoots.add(replaceNonAlphanumericByUnderscores(root));
```

Comment:
Not, these are short. These represent the namespace of each mapping like `quarkus.config`, `quarkus. opentelemetry`, etc.

Reply:
I can always provide a version which reuse the SrringBuilder which will be allocated later anyway...and we will save creating each time for each invocation of that method, wdyt? 
Unless are so few we really don't care to allocate twice (builders + strings)

Reply:
I think it is worth converting them all and doing the check later. I was using that method first and it seems to me it got a bit worse because you need to match each root to each env, but I didn't microbenchmark it.

Reply:
> I can always provide a version which reuse the SrringBuilder which will be allocated later anyway...and we will save creating each time for each invocation of that method, wdyt?

Probably it makes sense to have a version that takes a builder so we can reuse it.